src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java

```diff
@@ -528,6 +528,22 @@ public static boolean isCiphertext(byte[] bytes) {
                 });
     }
 
+    /**
+     * Re-key an EncryptedDocument to a new tenant without decrypting the document data. Decrypts the 
+     * document's encrypted document key (EDEK) then re-encrypts it to the new tenant. The DEK is then discarded.
+     * 
+     * @param encryptedDocument Document to re-key which includes encrypted bytes as well as EDEK.
+     * @param metadata          Metadata about the document being re-keyed.
+     * @param newTenantId       Tenant ID the document should be re-keyed to.
+     * @return                  EncryptedDocument that contains the new EDEK and unaltered encrypted fields.
+     */
+    public CompletableFuture<EncryptedDocument> rekeyDocument(EncryptedDocument encryptedDocument,
+            DocumentMetadata metadata, String newTenantId) {
+        return this.encryptionService.rekey(encryptedDocument.getEdek(), metadata, newTenantId).thenApply(newKey -> {
+            return new EncryptedDocument(encryptedDocument.getEncryptedFields(), newKey.getEdek());
```

Comment:
Minor, but you can use the "expression lambda" syntax and omit the extra curly braces and the `return` if you like.

Reply:
I think I'm doing this now